File: frontend/src/types/Account.ts
// Data coming from the backend
export type AccountResponse = {
  id: number;
  creatorId: number;
  platformUserId: string;
  platformUsername: string;
  profileLink?: string;
  displayName?: string;
  isActive: boolean;
  connectedAt?: string;
  disconnectedAt?: string;
  createdAt: string;
};

// Data sent when syncing/creating an account
export type AccountSyncRequest = {
  platformUserId: string;
  platformUsername: string;
  profileLink?: string;
  displayName?: string;
  isActive?: boolean;
  connectedAt?: string;
  disconnectedAt?: string;
};

File: backend/src/main/groovy/com/krasen/vizor/business/domain/Account.java
package com.krasen.vizor.business.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.OffsetDateTime;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
@Builder
public class Account {
    private Long id;
    private User creator;
    private String platformUserId;
    private String platformUsername;
    private String profileLink;
    private String displayName;
    private boolean isActive;
    private OffsetDateTime connectedAt;
    private OffsetDateTime disconnectedAt;
    private OffsetDateTime createdAt;
    private OffsetDateTime deletedAt;
}
